numpy/ma/core.py

```diff
@@ -2756,13 +2756,19 @@ def __new__(cls, data=None, mask=nomask, dtype=None, copy=False,
                     _data._sharedmask = True
         else:
             # Case 2. : With a mask in input.
-            # Read the mask with the current mdtype
-            try:
-                mask = np.array(mask, copy=copy, dtype=mdtype)
-            # Or assume it's a sequence of bool/int
-            except TypeError:
-                mask = np.array([tuple([m] * len(mdtype)) for m in mask],
-                                dtype=mdtype)
+            # If mask is boolean, create an array of True or False
+            if mask is True:
+                mask = np.ones(_data.shape, dtype=mdtype)
+            elif mask is False:
+                mask = np.zeros(_data.shape, dtype=mdtype)
```

Comment:
Maybe I should also check here that mdtype is dtype('bool') ? 
If I understand correctly mdtype can be a list for record arrays, in which case `mask = np.array(mask, copy=copy, dtype=mdtype)` throws a TypeError, then we go to the except below which suppose that mask is a list.
So in theory one should not use mask = True or False with a record array, or a list this case is not handled ?